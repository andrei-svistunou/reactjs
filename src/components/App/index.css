* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

html,
body {
  height: 100%;
}

.wrapper { 
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  width: 100%;
  max-width: 1024px;
  margin: 0 auto;
  font-family: Arial, sans-serif;
}

.panel {
  background-color: #f5f5f5;
  padding: 7px 50px;
}

.panel--empty {
  min-height: 20px;
}

.main {
  flex: 1;
  padding: 15px 25px;
}

.flex {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.upper-text {
  text-transform: uppercase;
}

.left-space-10 {
  margin-left: 10px;
}

.hide {
  display: none;
}