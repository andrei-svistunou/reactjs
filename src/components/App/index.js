import React, { Component } from 'react';
import { Hello, Img, TextField, WrapperBlock } from '../Task_1';
import { hot } from 'react-hot-loader'
import worldImg from '../../assets/img/world.jpg';
import './index.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      name: 'World',
      text: ''
    };
  }

  componentDidMount() {
    this.getMsg()
      .then(res => this.setState({ text: res.msg }))
      .catch(err => console.log(err));
  }

  async getMsg() {
    const response = await fetch('http://localhost:3003/');
    const body = await response.json();
    if (response.status !== 200) throw Error(body.message);
    return body;
  }

  render() {
    return (
      <div>
        {WrapperBlock}
        <Hello name={this.state.name} />
        <TextField text={this.state.text} />
        <Img imgSrc={worldImg} />
      </div>
    );
  }
}

export default hot(module)(App);
