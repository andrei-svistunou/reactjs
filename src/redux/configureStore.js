import devConfigStore from './configureStore.dev';
import prodConfigStore from './configureStore.prod';

const configStore = process.env.NODE_ENV === 'production' ? prodConfigStore : devConfigStore;

export default configStore;
