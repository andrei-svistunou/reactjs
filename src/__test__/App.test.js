import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import configureStore from '../redux/configureStore';
import App from '../components/App';

const initialState = window.REDUX_INITIAL_STATE || {};
const store = configureStore(initialState);

describe('<App />', () => {
  test('should render <App />', () => {
    const app = shallow(<Provider store={store}><App/></Provider>);

    expect(toJson(app)).toMatchSnapshot();
  });
});

