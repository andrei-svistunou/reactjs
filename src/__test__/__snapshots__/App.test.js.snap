// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render <App /> 1`] = `
<div
  className="wrapper"
>
  <Header
    clickHandler={[Function]}
    defaultStatus={false}
  >
    <Search
      clickHandler={[Function]}
      id="main-search"
    />
  </Header>
  <div
    className="panel flex  panel--empty"
  />
  <main
    className="main"
  >
    <NotFound
      text="No films found"
    />
  </main>
  <Footer>
    <Logo
      text="netflixroulett"
    />
  </Footer>
</div>
`;
